web/frontend/src/components/slam.vue

```diff
@@ -68,10 +74,18 @@ const fetchSLAMMap = (name: string): Promise<Uint8Array> => {
         reject(error);
       }
     });
-    getPointCloudMap.on('end', (end?: { code: number }) => {
-      if (end === undefined || end.code !== 0) {
-        // the error will be logged in the 'status' callback
-        return;
+    getPointCloudMap.on('end', (end?: { code: number, details: string, metadata: grpc.Metadata }) => {
```

Comment:
just for my knowledge, do you know what this `(end?: { ... })` syntax means? It looks like some form of [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining), but I don't quite get how it works here

Reply:
It means that end can be `null` 
https://stackoverflow.com/questions/37632760/what-is-the-question-mark-for-in-a-typescript-parameter-name#:~:text=Not%20exactly.-,Question%20mark%20means%20%22optional%22.,%7C%20undefined%20%3D%20undefined%22..

Reply:
nice, ty!

additionally spoke offline that we think based on talking with max on the sdk team that end would be null if the server didn't respond, but would be populated if the server responded with an error.

further, if the callback passed to getPointCloudMap.on calls neither reject nor resolve, then the logical thread of execution will hang at [this](https://github.com/viamrobotics/rdk/blob/6d097d9e2905c4ef9bedc9aa619aff3c4b38a39a/web/frontend/src/components/slam.vue#L146) line